File: ios/db_explorer/drivers/DriverManager.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(DriverManager, NSObject)

RCT_EXTERN_METHOD(send_message:(NSString *)data
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

@end

File: ios/db_explorer/drivers/DriverManager.swift
import Foundation

@objc(DriverManager)
class DriverManager : NSObject {
  @objc static func requiresMainQueueSetup() -> Bool {
      return false
  }
  
  @objc func postMessage(_ data: NSString, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) -> Void {
    DispatchQueue.global(qos: .background).async {
      let input = String(data)
      let response = receive_message(input)
      let result = String(cString: response!)
      free_message(UnsafeMutablePointer(mutating: response))
      DispatchQueue.main.async {
        resolve(result)
      }
    }
  }
}
